exchanges/graphcache/src/types.ts

```diff
@@ -136,22 +136,21 @@ type ResolverResult =
   | undefined;
 
 // Cache resolvers are user-defined to overwrite an entity field result
-export type Resolver = (
-  parent: Data,
-  args: Variables,
-  cache: Cache,
-  info: ResolveInfo
-) => ResolverResult;
+export type Resolver<
+  ParentData = Data,
+  Args = Variables,
+  Result = ResolverResult
+> = (parent: ParentData, args: Args, cache: Cache, info: ResolveInfo) => Result;
 
 export interface ResolverConfig {
   [typeName: string]: {
     [fieldName: string]: Resolver;
   };
 }
 
-export type UpdateResolver = (
-  result: Data,
-  args: Variables,
+export type UpdateResolver<Result = Data | Data[], Args = Variables> = (
```

Comment:
Not entirely sure about this change but imo it should be possible to return array data from a mutation, no?

Reply:
I could have sworn we had a type definition for this (basically JSON) in core - did we revert it?

Reply:
It will never return an array as `result` will always be the mutation data itself, never the item or field, since it imitates `parent` from GraphQL resolvers (like the other resolvers)

See: https://github.com/FormidableLabs/urql/blob/f6979d5be57c05a2c5ad911f9d4ca1501b7c688e/exchanges/graphcache/src/operations/write.ts#L292-L293